RecoLocalCalo/EcalRecAlgos/src/PulseChiSqSNNLS.cc

```diff
@@ -241,17 +272,18 @@ bool PulseChiSqSNNLS::Minimize(const SampleMatrix &samplecor, double pederr, con
   
 }
 
-bool PulseChiSqSNNLS::updateCov(const SampleMatrix &samplecor, double pederr, const FullSampleMatrix &fullpulsecov) {
+bool PulseChiSqSNNLS::updateCov(const SampleMatrix &samplecov, const FullSampleMatrix &fullpulsecov) {
  
   const unsigned int nsample = SampleVector::RowsAtCompileTime;
   const unsigned int npulse = _bxs.rows();
 
-  const double pederr2 = pederr*pederr;
-  _invcov = pederr2*samplecor; //
+  _invcov = samplecov; //
```

Comment:
it looks like this part and the way samplecov is now computed upstream lead to some differences at numerical precision level observed in DQM comparisons.

pederr2 was ``` (pederr*gainratio)*(pederr*gainratio) ```, while now it is ```(pederr* pederr* gainratio* gainratio)```

the other part of the logic that appears different is ``` abs(bx)>=100 ``` added in a few places, but IIUC, this never happens with the current defaults.
@bendavid 
please confirm